worlds/metroidprime/Logic.py

```diff
@@ -0,0 +1,429 @@
+    def can_heat(self, world: "MetroidPrimeWorld", state: CollectionState) -> bool:
+        if world.options.non_varia_heat_damage:
+            return state.has(SuitUpgrade.Varia_Suit.value, world.player)
+        else:
+            return state.has_any(HEAT_PROTECTION_REQUIREMENTS, world.player)
+
+    def can_phazon(self, world: "MetroidPrimeWorld", state: CollectionState) -> bool:
+        return state.has(SuitUpgrade.Phazon_Suit.value, world.player)
+
+    def has_energy_tanks(
+        self, world: "MetroidPrimeWorld", state: CollectionState, count: int
+    ) -> bool:
+        return state.has(SuitUpgrade.Energy_Tank.value, world.player, count)
+
+    def can_infinite_speed(
+        self, world: "MetroidPrimeWorld", state: CollectionState
+    ) -> bool:
+        return self.can_boost(world, state) and self.can_bomb(world, state)
+
+    def can_defeat_sheegoth(
+        self, world: "MetroidPrimeWorld", state: CollectionState
+    ) -> bool:
+        return (
+            self.can_bomb(world, state)
+            or self.can_missile(world, state, 1)
+            or self.can_power_bomb(world, state)
+            or self.can_plasma_beam(world, state)
+        )
+
+    def can_backwards_lower_mines(
+        self, world: "MetroidPrimeWorld", state: CollectionState
+    ) -> bool:
+        return bool(world.options.backwards_lower_mines.value)
+
+    def has_power_bomb_count(
+        self, world: "MetroidPrimeWorld", state: CollectionState, required_count: int
+    ) -> bool:
+        count = state.count(SuitUpgrade.Power_Bomb_Expansion.value, world.player)
+        if state.has(SuitUpgrade.Main_Power_Bomb.value, world.player):
+            count += 4
+        return count >= required_count
+
+    def can_warp_to_start(
+        self, world: "MetroidPrimeWorld", state: CollectionState
+    ) -> bool:
+        SAVE_ROOMS = [
+            RoomName.Landing_Site.value,
+            RoomName.Save_Station_1.value,
+            RoomName.Save_Station_2.value,
+            RoomName.Save_Station_3.value,
+            RoomName.Save_Station_Magmoor_A.value,
+            RoomName.Save_Station_Magmoor_B.value,
+            RoomName.Save_Station_A.value,
+            RoomName.Save_Station_B.value,
+            RoomName.Save_Station_C.value,
+            RoomName.Save_Station_D.value,
+            RoomName.Cargo_Freight_Lift_to_Deck_Gamma.value,
+            RoomName.Save_Station_Mines_A.value,
+            RoomName.Save_Station_Mines_B.value,
+            RoomName.Save_Station_Mines_C.value,
+        ]
+        for room in SAVE_ROOMS:
+            if state.can_reach_region(room, world.player):
+                return True
+        return False
+
+    def _can_combat_generic(
+        self,
+        world: "MetroidPrimeWorld",
+        state: CollectionState,
+        normal_tanks: int,
+        minimal_tanks: int,
+        requires_charge_beam: bool = True,
+    ) -> bool:
+        difficulty = CombatLogicDifficulty(world.options.combat_logic_difficulty)

```

Comment:
Again, checking options in rules is not ideal, but, at the very least, options used in rules should be using `.value`.
```suggestion
        difficulty = CombatLogicDifficulty(world.options.combat_logic_difficulty.value)
```